lib/validation.js

```diff
@@ -139,7 +139,8 @@ internals.input = function (source, request, next) {
 
     request._protect.run(postValidate, (exit) => {
 
-        return schema(request[source], localOptions, exit);
+        const bind = request.route.settings.bind || request.route.realm || null;
+        return schema.call(bind, request[source], localOptions, exit);
```

Comment:
Change the whole framework so you always pass a `context` object rather than relying on `this` 😄 

But seriously, yeah there really isn't a way to do that other than `call` here.